File: 신창호/Lv2/멀리 뛰기.java
class Solution {
    public long solution(int n) {
        // 1칸, 2칸 - >DP 문제
        long[] dp = new long[n+1];

        if(n >= 1) dp[1] = 1L ;
        if(n >= 2) dp[2] = 2L;
        for(int i = 3; i<= n; i++){
            dp[i] = (dp[i-1] + dp[i-2])%1234567;
        }

        long answer = dp[n];
        return answer;
    }
}

File: 조은정/Lv.2/멀리 뛰기.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;
typedef long long ll;

map<pair<int, int>, ll> mp;

ll combination(int u, int v)
{
    if (mp[{u, v}])
        return mp[{u, v}];

    if (u == v || v == 0)
        return 1;
    else
        return mp[{u, v}] = (combination(u - 1, v - 1) + combination(u - 1, v)) % 1234567;
}

ll solution(int n)
{
    ll answer = 0;

    for (int i = 0; i <= n / 2; ++i)
    { // 2칸 i개, 1칸 (n - i * 2)개
        answer = (answer + combination(n - i, i)) % 1234567;
    }

    return answer;
}